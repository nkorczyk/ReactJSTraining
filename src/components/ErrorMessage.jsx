// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {
  message: string,
}

const NotFoundContainer = styled.div`
  background: #fff;
  margin: 0 auto;
  max-width: 800px;
  max-height: 200px;
`;

const NotFoundMessage = styled.h3`
  padding: 80px;
  margin: 0 auto;
`;

const ErrorMessage = ({ message }: Props) => (
  <NotFoundContainer>
    <NotFoundMessage>{message}</NotFoundMessage>
  </NotFoundContainer>
);

export default ErrorMessage;
