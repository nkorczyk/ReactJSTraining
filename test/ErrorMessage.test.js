import React from 'react';
import { mount } from 'enzyme';
import ErrorMessage from "../src/components/ErrorMessage";

describe('should render ErrorMessage component', () => {
  const element = mount(<ErrorMessage message="Test message" />);
  test('Snapshot test with default props', () => {
    expect(element).toMatchSnapshot();
  });
});

test('should render error message', () => {
  const element = mount(<ErrorMessage message="Test message" />);
  expect(element.text()).toEqual('Test message');
});
