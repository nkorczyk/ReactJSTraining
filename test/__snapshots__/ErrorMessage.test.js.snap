// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ErrorMessage component Snapshot test with default props 1`] = `
<ErrorMessage
  message="Test message"
>
  <div
    className="not-found"
  >
    <h3
      className="not-found-message"
    >
      Test message
    </h3>
  </div>
</ErrorMessage>
`;
